airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py

```diff
@@ -0,0 +1,349 @@
+    def stream_slices(self) -> Iterable[StreamSlice]:
+        """
+        Add 'funnel_name' to the slice, the rest code is exactly the same as in super().stream_slices(...)
+        """
+        if not self.parent_stream_configs:
+            yield from []
+        else:
+            for parent_stream_config in self.parent_stream_configs:
+                parent_stream = parent_stream_config.stream
+                parent_field = parent_stream_config.parent_key.eval(self.config)  # type: ignore # parent_key is always casted to an interpolated string
+                partition_field = parent_stream_config.partition_field.eval(self.config)  # type: ignore # partition_field is always casted to an interpolated string
+                for parent_stream_slice in parent_stream.stream_slices(
+                    sync_mode=SyncMode.full_refresh, cursor_field=None, stream_state=None
+                ):
+                    empty_parent_slice = True
+                    parent_partition = parent_stream_slice.partition if parent_stream_slice else {}
+
+                    for parent_record in parent_stream.read_records(
+                        sync_mode=SyncMode.full_refresh, cursor_field=None, stream_slice=parent_stream_slice, stream_state=None
+                    ):
+                        # Skip non-records (eg AirbyteLogMessage)
+                        if isinstance(parent_record, AirbyteMessage):
+                            if parent_record.type == Type.RECORD:
+                                parent_record = parent_record.record.data
+                            else:
+                                continue
+                        elif isinstance(parent_record, Record):
+                            parent_record = parent_record.data
+                        try:
+                            partition_value = dpath.util.get(parent_record, parent_field)
+                        except KeyError:
+                            pass
+                        else:
+                            empty_parent_slice = False
+                            yield StreamSlice(
+                                partition={
+                                    partition_field: partition_value,
+                                    "funnel_name": parent_record.get("name"),
+                                    "parent_slice": parent_partition,
+                                },
+                                cursor_slice={},
+                            )
+                    # If the parent slice contains no records,
+                    if empty_parent_slice:
+                        yield from []
+
+
+@dataclass
+class EngagePaginationStrategy(PageIncrement):
+    """
+    Engage stream uses 2 params for pagination:
+    session_id - returned after first request
+    page - incremental page number
+    """
+
+    def next_page_token(self, response, last_records: List[Mapping[str, Any]]) -> Optional[Mapping[str, Any]]:
```

Comment:
can you create an issue for implementing this pattern in the CDK?

Reply:
I will add it to lour list of custom components

Reply:
bump on this request. I don't see this component in [the feature request spreadsheet ](https://docs.google.com/spreadsheets/d/1VReV77rAa4TjLPFksfPFFTvvA3fsy2Ps7kvnLyq1JEs/edit?usp=sharing)